packages/next-on-pages/src/buildApplication/buildApplication.ts

```diff
@@ -145,3 +151,27 @@ async function prepareAndBuildWorker(
 
 	cliSuccess(`Generated '${outputtedWorkerPath}'.`);
 }
+
+/**
+ * Copies the no nodejs_compat static error page to the output directory
+ *
+ * @param outputDir path of the output directory
+ * @returns the filename of the html file copied
+ */
+async function copyNoNodejsCompatStaticErrorPage(
+	outputDir: string,
+): Promise<string> {
+	const htmlFileName = 'no-nodejs-compat-flag-error-page.html';
+	const noNodejsCompatFlagStaticErrorPagePath = join(
+		__dirname,
+		'..',
+		'no-nodejs-compat-flag-static-error-page',
+		'dist',
+		'index.html',
+	);
+	await cp(
+		noNodejsCompatFlagStaticErrorPagePath,
+		join(outputDir, htmlFileName),
```

Comment:
I'd consider putting this into `${outputDir}/cdn-cgi/errors/no-nodejs_compat.html` or something like that rather than just in the root of the build output directory, just so it's tucked away in something we clearly own.

Reply:
yeah I totally agree that we should make it clear that it's something from us, I thought it was clear enough given the file name, but yeah putting it into `cdn-cgi` (which, since you're suggesting it I guess is allowed/doesn't generate conflicts/issues) might just be cleaner simpler, I'll look into that, thanks! 😄

Reply:
https://developers.cloudflare.com/fundamentals/reference/cdn-cgi-endpoint/

This will actually mean that this will be inaccessible for anyone trying to directly access this asset and it can effectively only retrieved by the Pages Functions Worker :)